compio-driver/src/key.rs

```diff
@@ -1,44 +1,187 @@
-/// A typed wrapper for key of Ops submitted into driver
+use std::{io, marker::PhantomData, pin::Pin, task::Waker};
+
+use compio_buf::BufResult;
+
+use crate::{OpCode, Overlapped, PushEntry, RawFd};
+
+/// An operation with other needed information. It should be allocated on the
+/// stack. The pointer to this struct is used as user_data, and on Windows, it
+/// is used as the pointer to OVERLAPPED.
+///
+/// Convert any user_data to *const RawOp<()> is valid. Then it could be
```

Comment:
Don't say convert from user_data. user_data is not some specific type, it's merely an usize with a specific variable name.

Say something like:

`*const RawOp<dyn OpCode>` can be obtained from any `Key<T: OpCode>` by first casting `Key::user_data` to `*const RawOp<()>`, then upcasted with `upcast_fn`.

Also, mention the layout and why cast like so is safe.

Reply:
About the cast, it is mentioned in `as_opaque*`